tests/CppWinRTTests.cpp

```diff
@@ -646,3 +647,26 @@ TEST_CASE("CppWinRTTests::ResumeForegroundTests", "[cppwinrt]")
     }().get();
 }
 #endif // coroutines
+
+TEST_CASE("CppWinRTTests::ThrownExceptionWithMessage", "[cppwinrt]")
+{
+    SetRestrictedErrorInfo(nullptr);
+    []()
+    {
+        try
+        {
+            throw winrt::hresult_access_denied(L"Puppies not allowed");

```

Comment:
The origination from cppwinrt can vary based on the mechanism through which it is thrown (although on cpp20 and post-September-2022 it should be far more consistent).  Consider also validating `winrt::check_hresult(E_ACCESSDENIED, L"Puppies are wonderful");`